### PR TITLE
Unify reduced lung test mesh construction

### PR DESCRIPTION
**Pull request by maxiludwig**
## Description and Context

The junction and boundary condition tests built their meshes from `Discret::Elements::RedAirway`, which they only used as a generic two-node line element.
They now share the mesh helpers of `4C_reduced_lung_helpers_test.cpp`, which move into the shared test utils and build via `build_discretization_from_nodes_and_elements` like the production code does.
This removes an undeclared dependency of the reduced lung tests on `red_airways`, along with three near-duplicate mesh helpers.

## Disclosure of AI assistance

The changes in this PR were made by Claude Opus 5.

### CODE REVIEW

File: src/reduced_lung/tests/4C_reduced_lung_test_utils_test.hpp [hunk 1]
```diff
@@ -13,19 +13,76 @@
...
 #include "4C_linalg_vector.hpp"
+#include "4C_rebalance.hpp"
 #include "4C_reduced_lung_airways.hpp"
+#include "4C_reduced_lung_helpers.hpp"
 #include "4C_reduced_lung_terminal_unit.hpp"
 
+#include <mpi.h>
+
+#include <array>
+#include <memory>
```
Comment (davidrudlstorfer) [inline src/reduced_lung/tests/4C_reduced_lung_test_utils_test.hpp:28]: Do we really need these imports here?

**Review by davidrudlstorfer — APPROVED** [review]

File: src/reduced_lung/tests/4C_reduced_lung_test_utils_test.hpp (hunk 1 above)
Comment (maxiludwig) [reply src/reduced_lung/tests/4C_reduced_lung_test_utils_test.hpp:28]: They all are used, but they would also come with the included header files.

**Review by lkoeglmeier — APPROVED** [review]
